ports/psoc6/machine_spi.c

```diff
@@ -65,97 +119,70 @@ static inline void spi_obj_free(machine_spi_obj_t *spi_obj_ptr) {
-                mode = CYHAL_SPI_MODE_00_LSB;
-            } else {
-                mode = CYHAL_SPI_MODE_01_LSB;
-            }
-        } else {
-            if (phase == 0) {
-                mode = CYHAL_SPI_MODE_10_LSB;
-            } else {
-                mode = CYHAL_SPI_MODE_11_LSB;
-            }
-        }
-    } else {
-        if (polarity == 0) {
-            if (phase == 0) {
-                mode = CYHAL_SPI_MODE_00_MSB;
-            } else {
-                mode = CYHAL_SPI_MODE_01_MSB;
-            }
-        } else {
-            if (phase == 0) {
-                mode = CYHAL_SPI_MODE_10_MSB;
-            } else {
-                mode = CYHAL_SPI_MODE_11_MSB;
-            }
-        }
-    }
-    return mode;
+static inline void spi_init(machine_spi_obj_t *machine_spi_obj, int spi_mode) {
+    cyhal_spi_mode_t mode = spi_mode_select(machine_spi_obj->firstbit, machine_spi_obj->polarity, machine_spi_obj->phase);
+    // set the baudrate
+    cyhal_spi_set_frequency(&machine_spi_obj->spi_obj, machine_spi_obj->baudrate);
+    // Initialise the SPI peripheral if any arguments given, or it was not initialised previously.
+    cy_rslt_t result = cyhal_spi_init(&machine_spi_obj->spi_obj, machine_spi_obj->mosi->addr, machine_spi_obj->miso->addr, machine_spi_obj->sck->addr, machine_spi_obj->ssel->addr, NULL, machine_spi_obj->bits, mode, spi_mode);
+    spi_assert_raise_val("SPI initialisation failed with return code %x !", result);
 }
 
 static void machine_spi_print(const mp_print_t *print, mp_obj_t self_in, mp_print_kind_t kind) {
     machine_spi_obj_t *self = MP_OBJ_TO_PTR(self_in);
-    mp_printf(print, "SPI(id=%u, baudrate=%u, polarity=%u, phase=%u, bits=%u, firstbit=%u, sck=%d, mosi=%d, miso=%d)",
-        self->spi_id, self->baudrate, self->polarity,
+    mp_printf(print, "SPI(baudrate=%u, polarity=%u, phase=%u, bits=%u, firstbit=%u, ssel=%d, sck=%d, mosi=%d, miso=%d)",
+        self->baudrate, self->polarity,
+        self->phase, self->bits, self->firstbit,
+        self->ssel->addr, self->sck->addr, self->mosi->addr, self->miso->addr);
+}
+
+
+static void machine_spi_slave_print(const mp_print_t *print, mp_obj_t self_in, mp_print_kind_t kind) {
+    machine_spi_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    mp_printf(print, "SPISlave(baudrate=%u, polarity=%u, phase=%u, bits=%u, firstbit=%u, ssel=%d, sck=%d, mosi=%d, miso=%d)",
+        self->baudrate, self->polarity,
         self->phase, self->bits, self->firstbit,
-        self->sck->addr, self->mosi->addr, self->miso->addr);
+        self->ssel->addr, self->sck->addr, self->mosi->addr, self->miso->addr);
 }
 
-mp_obj_t machine_spi_make_new(const mp_obj_type_t *type, size_t n_args, size_t n_kw, const mp_obj_t *all_args) {
-    enum { ARG_id, ARG_baudrate, ARG_polarity, ARG_phase, ARG_bits, ARG_firstbit, ARG_sck, ARG_mosi, ARG_miso };
+mp_obj_t machine_spi_master_init_helper(machine_spi_obj_t *self, size_t n_args, size_t n_kw, const mp_obj_t *all_args) {
```

Comment:
Mmm, do we really need to machine_spi_xxxx_init_helpers()? We had one only before, no?
Can we just not have one? and if the user pass the ssel for master complain, and if he does not pass it for slave also complain?
Other than that parameter, the rest is the same or?